crud/common/sharding_key.lua

```diff
@@ -115,6 +116,87 @@ function sharding_key_module.fetch_on_router(space_name)
     return cache.sharding_key_def
 end
 
+-- Make sure sharding key definition is a part of primary key.
+function sharding_key_module.is_part_of_pk(space_name, space_format,
+                                           primary_index, sharding_key_def)
+    dev_checks('string', 'table', 'table', 'table')
+
+    if cache.as_index_object[space_name] == nil then
+        cache.as_index_object[space_name] =
+            sharding_key_module.as_index_object(space_format, space_name,
+                                                sharding_key_def)
+    end
+    local sharding_key_as_index_obj = cache.as_index_object[space_name]
```

Comment:
I guess it is 'sharding key definition as index object', not 'sharding key as index object'. Maybe not so big problem, a bit confusing.

Reply:
rename `as_index_object` to `keydef_as_index_object`